src/status_im/ui/screens/accounts/events.cljs

```diff
@@ -158,13 +159,18 @@
   "Takes effects (containing :db) + new account fields, adds all effects necessary for account update."
   [{{:accounts/keys [accounts current-account-id] :as db} :db :as fx} new-account-fields]
   (let [current-account (get accounts current-account-id)
-        new-account     (merge current-account new-account-fields)]
-    (-> fx
-        (assoc-in [:db :accounts/accounts current-account-id] new-account)
-        (assoc ::save-account new-account
-               ::broadcast-account-update (merge (select-keys db [:current-public-key :web3])
-                                                 (select-keys new-account [:name :photo-path :status
-                                                                           :updates-public-key :updates-private-key]))))))
+        new-account     (merge current-account new-account-fields)
+        broadcast-fields [:name :photo-path :status
+                          :updates-public-key :updates-private-key]]
+    (cond-> fx
+            true
+            (assoc-in [:db :accounts/accounts current-account-id] new-account)
+            true
+            (assoc ::save-account new-account)
+
+            (not (empty? (clojure.set/intersection (set (keys new-account-fields)) (set broadcast-fields))))
```

Comment:
I don't know what's the consensus of `seq` vs `not-empty`, but depending on style preferences this could be:
```
(not-empty (clojure.set/intersection (set (keys new-account-fields)) (set broadcast-fields))
```

or (preferred by clojure style guides)

```
 (seq (clojure.set/intersection (set (keys new-account-fields)) (set broadcast-fields)))
```

Reply:
it might also be worth looking into using of `merge-fx` here, we could `re-factor the broadcast-account-updates` bit into his own function, which returns the `fxs` or nothing and then we can get rid of `cond->` and thread straight through.

Reply:
thanks, for some reason `seq` is only one thing I can't get used, but i agree better to use it here